tools/toolchains/arm.py

```diff
@@ -298,12 +301,21 @@ def __init__(self, target, *args, **kwargs):
         elif target.core == "Cortex-M7FD":
             self.flags['common'].append("-mfpu=fpv5-d16")
             self.flags['common'].append("-mfloat-abi=softfp")
+        elif target.core.startswith("Cortex-M23"):
+            self.flags['common'].append("-march=armv8-m.base")
+        elif target.core.startswith("Cortex-M33"):
+            self.flags['common'].append("-march=armv8-m.main")
+
+        if target.core == "Cortex-M23" or target.core == "Cortex-M33":
+            self.flags['common'].append("-mcmse")
```

Comment:
What does this flag do?

Reply:
Cortex-M23-NS  : Non-secure build
Cortex-M23: Secure build (-mcmse flag is required to build a secure image).

Reply:
Super helpful.

Reply:
> Super helpful.

It might be better if this type of flags are part of the commit message what they do or in the code as comment if they have special meaning (why are we enabling a secure image).